lib/connections/limiter.go

```diff
@@ -162,3 +263,37 @@ func (b *atomicBool) set(v bool) {
 func (b *atomicBool) get() bool {
 	return atomic.LoadInt32((*int32)(b)) != 0
 }
+
+// Utility functions for atomic operations on device limiters map
+func (lim *limiter) getOrCreateDeviceWriteLimiter(deviceID protocol.DeviceID) (*rate.Limiter, bool) {
```

Comment:
Personally I would drop these getOrCreate functions and do what is necessary directly where called. This is just in two places, and in one place you already know that no device limiter exists yet. However that's a style question and I am quite the contrary of an authority on style, so don't act on my suggestion unless somebody else (or you yourself) agree with it.